documentation/modules/configuring/con-config-node-pools.adoc

```diff
@@ -0,0 +1,105 @@
+// Module included in the following assemblies:
+//
+// assembly-config.adoc
+
+[id='config-node-pools-{context}']
+= (Preview) Configuring node pools
+
+[role="_abstract"]
+Update the `spec` properties of the `KafkaNodePool` custom resource to configure a node pool deployment.
+
+NOTE: The node pools feature is available as a preview. Node pools are not enabled by default, so you must xref:ref-operator-kafka-node-pools-feature-gate-{context}[enable the `KafkaNodePools` feature gate] before using them. 
+
+A node pool refers to a distinct group of Kafka nodes within a Kafka cluster.
+Each pool has its own unique configuration, which includes mandatory settings for the number of replicas, roles, and storage allocation.
+
+Optionally, you can also specify values for the following properties:
+
+* `resources` to specify memory and cpu requests and limits
+* `template` to specify custom configuration for pods and other Kubernetes resources
+* `jvmOptions` to specify custom JVM configuration for heap size, runtime and other options 
+
+The `Kafka` resource represents the configuration for all nodes in the Kafka cluster.
+The `KafkaNodePool` resource represents the configuration for nodes only in the node pool.
+Configuration options not specified in the `KafkaNodePool` resource are inherited from the `Kafka` resource.
+Configuration specified in the `KafkaNodePool` resource takes precedence if set in both resources
+For example, if `jvmOptions` are specified in the node pool configuration and Kafka configuration, the node pool configuration is used.
```

Comment:
Can we go into more details and explain that the options are not combined? To better explain what is covered in this part of the proposal:

> When the optional field is not present in `KafkaNodePool` and is present in `Kafka.spec.kafka`, it will default to the configured value from the `Kafka` CR.
> So when `KafkaNodePool.spec.<field>` exists, `Kafka.spec.kafka.<field>` will be ignored.
> This will be applied only on the highest level as demonstrated in the example below:
> 
> * If the `KafkaNodePool.spec.jvmOptions` does not exist but `Kafka.spec.kafka.jvmOptions` does exist, the values from `Kafka.spec.kafka.jvmOptions` will be used.
> * When `KafkaNodePool.spec.jvmOptions` exists and has `-Xmx: 1024m` and `Kafka.spec.kafka.jvmOptions` exists as well with `-Xms: 512m`, the operator will completely ignore the value from `Kafka.spec.kafka.jvmOptions` and use only the value from `KafkaNodePool.spec.jvmOptions`.
> * When `KafkaNodePool.spec.template` contains only the field `podSet.metadata.labels`, the operator will completely ignore the corresponding values from `Kafka.spec.kafka.template` (for example `podSet.metadata.annotations`, but also for example `pod.metadata.labels`).